Source/IBMCloudAppID/internal/Utils.swift

```diff
@@ -252,5 +252,20 @@ public class Utils {
         return url.query?.components(separatedBy: "&").filter({(item) in item.hasPrefix(paramName)}).first?.components(separatedBy: "=")[1]
     }
 
+    public static func generateStateParameter(of length: Int) -> String {
+
+        let allowedChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+        let allowedCharsCount = allowedChars.count
+        var randomString = ""
+
+        for _ in 0..<length {
+            let randomNum = Int(arc4random_uniform(UInt32(allowedCharsCount)))
+            let randomIndex = allowedChars.index(allowedChars.startIndex, offsetBy: randomNum)
```

Comment:
Why do we have allowed characters? why are some characters not allowed? We should be generating a nonce and then doing base64 encoding on the nonce.